### ISSUE TITLE
no-undef false positive for $localize tags in component code

### ISSUE THREAD

**Issue by fogshot** [angular-eslint/angular-eslint#789]
I have migrated to eslint with an existing angular application. After some woes, it seemed to be working great. However, today I added my first i18n tag in component code. It seems that eslint does not recognize the `$localize` tag and throws the no-undef error. My configuration seems correct to me (see below). 

Could this be a bug with one of the angular-eslint plugins?

```JSON
{
  "root": true,
  "extends": [
    "eslint:recommended"
  ],
  "env": {
    "browser": true,
    "node": true,
    "amd": true,
    "jasmine": true,
    "es2021": true
  },
  "ignorePatterns": [
    "projects/**/*"
  ],
  "overrides": [
    {
      "files": [
        "*.ts"
      ],
      "parserOptions": {
        "project": [
          "tsconfig.json"
        ],
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@angular-eslint/recommended",
        "plugin:@angular-eslint/template/process-inline-templates"
      ],
      "rules": {
        "@angular-eslint/directive-selector": [
          "error",
          {
            "type": "attribute",
            "prefix": "app",
            "style": "camelCase"
          }
        ],
        "@angular-eslint/component-selector": [
          "error",
          {
            "type": "element",
            "prefix": "app",
            "style": "kebab-case"
          }
        ],
        // note you must disable the base rule as it can report incorrect errors
        "no-unused-vars": "off",
        "@typescript-eslint/no-unused-vars": [
          "error"
        ]
      }
    },
    {
      "files": [
        "*.html"
      ],
      "extends": [
        "plugin:@angular-eslint/template/recommended"
      ],
      "rules": {}
    }
  ]
}
```

```TS
this.snackBar.open(x.message, $localize`OK`);
```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/eslint-plugin`    | `^12.6.1` |
| `@typescript-eslint/parser`        | `^4.15.0` |
| `ESLint`                           | `^7.26.0` |
| `node`                             | `14.17.1` |

```sh
Angular CLI: 12.2.13
Node: 14.17.1
Package Manager: npm 7.20.1
OS: darwin x64

Angular: 12.2.13
... animations, cli, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.1202.13
@angular-devkit/build-angular     12.2.13
@angular-devkit/core              12.2.13
@angular-devkit/schematics        12.2.13
@angular/cdk                      12.2.12
@angular/material                 12.2.12
@angular/material-luxon-adapter   12.2.12
@schematics/angular               12.2.13
rxjs                              6.6.7
typescript                        4.3.5
```

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
Hi @fogshot, ESLint doesn’t have a way to know you are using a library which is creating a new global ($localize) so you need to tell it via your config or inline in the files via comments: https://eslint.org/docs/user-guide/configuring/language-options#specifying-globals

**Comment by fogshot** [comment]
Cheers @JamesHenry, that sorted it! My bad for not realizing this and thank you for educating me! :)

**Comment by JamesHenry** [comment]
All good @fogshot! Glad it's working for you
